Tests/ClientTests/Coordinators/CoordinatorTests.swift

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+@testable import Client
+
+final class CoordinatorTests: XCTestCase {
+    var navigationController: NavigationController!
+    var router: MockRouter!
+
+    override func setUp() {
+        super.setUp()
+        navigationController = MockNavigationController()
+        router = MockRouter(navigationController: navigationController)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        navigationController = nil
+        router = nil
+    }
+
+    func testAddChild() {
+        let subject = BaseCoordinator(router: router)
+        let child = BaseCoordinator(router: router)
+        subject.addChild(child)
+
+        XCTAssertEqual(subject.childCoordinators[0].id, child.id)
+    }
+
+    func testRemoveChild() {
+        let subject = BaseCoordinator(router: router)
+        let child = BaseCoordinator(router: router)
+        subject.addChild(child)
+        subject.removeChild(child)
+
+        XCTAssertTrue(subject.childCoordinators.isEmpty)
+    }
+}
+
+class MockRouter: NSObject, Router {
+    var navigationController: NavigationController
+    var rootViewController: UIViewController?
+
+    init(navigationController: NavigationController) {
+        self.navigationController = navigationController
+        super.init()
+    }
+
+    var presentCalled = 0
+    func present(_ viewController: UIViewController, animated: Bool, completion: (() -> Void)?) {
+        presentCalled += 1
+    }
```

Comment:
Minor nit:
I can understand why this is organized like this.... but it's deeply distressing to me... and also against the standard patterns we use in our app for organizing code.

Reply:
Oh you mean putting the var on top of the function? I find it easier to find things for tests that way. I can change it if you wish, although I'm guilty of having done that in many places in our project 😅

Reply:
Yeah, no need to change it, I'm not a big stickler for this, especially in tests.

I just wanted to share how I felt about it. :)

Reply:
I'm inclined to agree with Roux on this one. It's easier for you to read but harder for everyone else because it's not the standard convention across the project, so things aren't where people expect to find them. Only a minor concern tho so I also won't request it to be changed.